.ci/Jenkinsfile

```diff
@@ -0,0 +1,94 @@
+#!groovy
+/**
+ * Jenkins pipeline to build Corda DJVM
+ */
+
+/**
+ * Kill already started job.
+ * Assume new commit takes precendence and results from previous
+ * unfinished builds are not required.
+ * This feature doesn't play well with disableConcurrentBuilds() option
+ */
+@Library('existing-build-control')
+import static com.r3.build.BuildControl.killAllExistingBuildsForJob
+killAllExistingBuildsForJob(env.JOB_NAME, env.BUILD_NUMBER.toInteger())
+
+/**
+ * Sense environment
+ */
+boolean isPR = (env.BRANCH_NAME.startsWith('PR'))
+boolean isRelease = (env.TAG_NAME =~ /^release\/.+$/ || env.BRANCH_NAME =~ /^release\/.+$/)
+
+pipeline {
+    agent {
+        label 'standard'
+    }
+
+    options {
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '7', artifactDaysToKeepStr: '7'))
+        timeout(time: 3, unit: 'HOURS')
+    }
+
+    environment {
+        ARTIFACTORY_BUILD_NAME = "DJVM / Publish / Publish to Artifactory".replaceAll("/", "::")
+    }
+
+    stages {
+
+        stage('Build DJVM') {
+            steps {
+                sh "./gradlew --daemon --no-build-cache clean build"
+            }
+        }
+
+        stage('Build DJVM Example') {
+            steps {
+                dir('djvm-example') {
+                    sh "./gradlew --daemon --no-build-cache clean build"
+                }
+            }
+        }
+
+        stage('Publish to Artifactory') {
+            when {
+                expression { !isPR }
```

Comment:
This is wrong. The expression should say when you *do* want to publish, instead when you *don't* want to publish.
Consider a scenario of a feature branch, or any random branch which is not the master branch.